src/core/providers/ept/qgseptpointcloudindex.cpp

```diff
@@ -229,4 +210,46 @@ QgsCoordinateReferenceSystem QgsEptPointCloudIndex::crs() const
   return QgsCoordinateReferenceSystem::fromWkt( mWkt );
 }
 
+bool QgsEptPointCloudIndex::loadHierarchy()
```

Comment:
Hmm we should do lazy loading of the hierarchy - only load the necessary parts of the hierarchy and load deeper parts when needed - large datasets may contain hundreds/thousands of files. Especially later when we support datasets through HTTP, this may take far too long... Or shall we defer that for later when we add support remote datasets?

Reply:
lets defer if for now..